frontend/src/components/review/CommentsPanel.tsx

```diff
@@ -113,27 +113,32 @@ const CommentsPanel = ({
 
   const CommentsList = () => {
     if (comments.length > 0) {
-      return (
-        <Box>
-          {/* TODO: show correct placement of the WIPComment component once existing comment is displayed */}
-          {comments.map((comment: CommentResponse, i: number) => (
-            <ExistingComment
-              key={comment.id}
-              commentsStateIdx={i}
-              userName={
-                translatorId === comment.userId ? translatorName : reviewerName
-              }
+      const commentThreads = [];
+      for (let i = 0; i < comments.length; i += 1) {
```

Comment:
This loop creates the comment threads by going through the `comments` array and creating a `CommentThread` for each `threadHead` comment. Any non `threadHead` comments following a `threadHead` comment are passed to the `CommentThread` as `replies`. Please let me know if there is a nicer way to do this.